Sources/Plasma/Apps/plClient/Mac-Cocoa/main.mm

```diff
@@ -0,0 +1,513 @@
+        //if our working path is inside our bundle - get out and
+        //point it to the containing folder
+        if([[[NSFileManager defaultManager] currentDirectoryPath] isEqualToString:@"/"]) {
+            chdir([[currentPath stringByDeletingLastPathComponent] cStringUsingEncoding:NSUTF8StringEncoding]);
+        }
+    }
+    
+    plFileName serverIni = "server.ini";
+    if (cmdParser.IsSpecified(kArgServerIni))
+        serverIni = cmdParser.GetString(kArgServerIni);
+    
+    FILE *serverIniFile = plFileSystem::Open(serverIni, "rb");
+    if (serverIniFile)
+    {
+        fclose(serverIniFile);
+        pfConsoleEngine tempConsole;
+        tempConsole.ExecuteFile(serverIni);
+    }
+    else
+    {
+        hsMessageBox("No server.ini file found.  Please check your URU installation.", "Error", hsMessageBoxNormal);
+        [NSApplication.sharedApplication terminate:nil];
+    }
+    
+    if (cmdParser.IsSpecified(kArgLocalData))
+    {
+        gDataServerLocal = true;
+        gSkipPreload = true;
+    }
+    if (cmdParser.IsSpecified(kArgSkipPreload))
+        gSkipPreload = true;
+    
+#ifndef PLASMA_EXTERNAL_RELEASE
+    //if (cmdParser.IsSpecified(kArgSkipLoginDialog))
+    //    doIntroDialogs = false;
+    if (cmdParser.IsSpecified(kArgPlayerId))
+        NetCommSetIniPlayerId(cmdParser.GetInt(kArgPlayerId));
+    if (cmdParser.IsSpecified(kArgStartUpAgeName))
+        NetCommSetIniStartUpAge(cmdParser.GetString(kArgStartUpAgeName));
+    
+    plPipeline::fInitialPipeParams.TextureQuality = 2;
+    //if (cmdParser.IsSpecified(kArgPvdFile))
+     //   plPXSimulation::SetDefaultDebuggerEndpoint(cmdParser.GetString(kArgPvdFile));
+    //if (cmdParser.IsSpecified(kArgRenderer))
+    //    gClient.SetRequestedRenderingBackend(ParseRendererArgument(cmdParser.GetString(kArgRenderer)));
+#endif
+    
+    NetCommStartup();
+    NetCommConnect();
+    [[PLSServerStatus sharedStatus] loadServerStatus];
+    
+    if(gSkipPreload) {
+        [self initializeClient];
+    } else {
+        [self prepatch];
+    }
+}
+
+- (void) prepatch {
+    self.patcherWindow = [[PLSPatcherWindowController alloc] initWithWindowNibName:@"PLSPatcherWindowController"];
+    self.patcher = [PLSPatcher new];
+    self.patcher.delegate = self;
+    
+    self.currentModalSession = [NSApp beginModalSessionForWindow:self.patcherWindow.window];
+    [NSApp runModalSession:self.currentModalSession];
+    [self.patcherWindow.window center];
+    [self.patcher start];
+}
+
+- (void) showLoginWindow {
+    self.loginWindow = [[PLSLoginWindowController alloc] init];
+    self.loginWindow.delegate = self;
+    [self.loginWindow showWindow:self];
+    [self.loginWindow.window makeKeyAndOrderFront:self];
+}
+
+- (void) initializeClient {
+    gClient.Init();
+    
+    // We should quite frankly be done initing the client by now. But, if not, spawn the good old
+    // "Starting URU, please wait..." dialog (not so yay)
```

Comment:
The splash dialog doesn't exist in the Mac client, right?

```suggestion
    // We should quite frankly be done initing the client by now. But, if not, spawn the good old
    // "Starting URU, please wait..." dialog (not so yay)
```

Reply:
Yeah - the Mac client has had a bit of a journey. There are still bits from @dpogue's original Mac GL client, which is roughly based on the Windows client. Some of these comments were handy to keep around because I could 1:1 map the intentions of the Mac client with sections of the Windows client. But as we figured out more and more what the Mac client should be some comments ended up mapping to nowhere. I'll do a cleanup pass.